pkg/gzip/options.go

```diff
@@ -100,9 +100,9 @@ func (o *Options) validate(moreArgs bool) error {
 	// error.
 	stat, _ := os.Stdin.Stat()
 
-	// No files passed and arguments and Stdin piped data found.
+	// No files passed and no arguments, Stdin piped data found.
 	// Stdin piped data is ignored if arguments are found.
-	if !moreArgs && (stat.Mode()&os.ModeNamedPipe) != 0 {
+	if !moreArgs && (stat.Mode()&os.ModeCharDevice) == 0 {
```

Comment:
another question here. What is the reason you can't just do:
if ! moreArgs {

char devices can serve as sources of a cpio stream. This is the fundamental idea of unix: that anything can supply you with data. So what's the point of this test?

Reply:
Good idea, seems to work correctly with this easier check

Reply:
Sorry, I couldn't get this to work without checking if something was on stdin. Without this we assume something is stdin but we don't test. So you would get an error if you just ran the `gzip` binary like the following. 
```
 % go run gzip.go
open /dev/stdin: permission denied
```

Reply:
I'm not seeing that problem at all.

I remove the stat and the test and it works fine.

can you strace and see why it's failing?

Reply:
I mean, all that said, what this code is doing is not really what we ought to do. We should just be using os.Stdin if there is no argument, not opening /dev/stdin, as that is a very linux-specific name. I hate to put this on you but would you be interested in removing the unnecessary linux-isms from gzip? If not I guess we can take your change but we'll need to file an issue against the package, it could use the help.